loadtests/loadtest/__init__.py

```diff
@@ -57,18 +63,49 @@ def __init__(self, *args, **kwargs):
 
             This method is called as many times as number of users.
         """
+        self.random_user = uuid.uuid4().hex
         super(TestBasic, self).__init__(*args, **kwargs)
         self.conf = Config(self.config['config']).get_map('loads')
-        if self.conf.get('smoke', False):
-            self.random_user = "test@restmail.net"
-            self.auth = RawAuth("Bearer %s" % self.conf.get('token'))
+        self.smoke = self.conf.get('smoke', False)
+        if self.smoke:
+            self.auth = self._get_fxa_auth()
         else:
-            self.random_user = uuid.uuid4().hex
             self.auth = HTTPBasicAuth(self.random_user, 'secret')
 
         # Create at least some records for this user
         self.nb_initial_records = random.randint(3, 100)
 
+    def __del__(self):
+        if self.smoke:
+            self._acct.clear()
+            self._client.destroy_account(self.user_email, self.random_user)
+
+    def _get_fxa_auth(self):
+        self.user_email = "rl-%s@restmail.net" % self.random_user
+        self._acct = TestEmailAccount(self.user_email)
+        self._client = Client(self.conf.get('fxa-url', DEFAULT_FXA_URL))
+        # Use userid as the password.
+        self._fxa_session = self._client.create_account(
+            self.user_email,
+            self.random_user
+        )
+
+        # Verify the account using the code from email.
+        self._acct.fetch()
+        for m in self._acct.messages:
+            if "x-verify-code" in m["headers"]:
+                self._fxa_session.verify_email_code(
+                    m["headers"]["x-verify-code"])
+
+        # Once the account verified, trade the assertion with a bearer token.
+        client = fxa.oauth.Client(client_id=self.conf['fxa-client-id'])
```

Comment:
ISTM there may be a potential mis-match between auth-server and oauth-server URLs here, e.g. using the prod auth server with stage oauth server.  Perhaps this is the cause of the "invalid assertion" errors you were seeing?